src/auth.rs

```diff
@@ -31,18 +33,25 @@ pub struct Authenticated<T> {
 
 impl<T: AuthScheme> AuthState for Authenticated<T> {}
 impl<T: AuthScheme> private::AuthState for Authenticated<T> {
-    fn maybe_credentials(&self) -> Option<ClientCredentials> {
-        Some(self.auth_scheme.credentials())
+    fn maybe_auth_header(&self) -> Option<AuthHeaderEntry> {
+        Some(self.auth_scheme.get_auth_header())
     }
 }
 
 #[derive(Debug, Clone)]
```

Comment:
`#[derive(Debug)]` would represent this as `.. { scheme: Plain(..) }` which does well to inform about how the auth scheme was provided, but not very informative for what type of auth scheme it is.

Preferably, we should have `AuthKey::Plain(..)`